### DISCUSSION TITLE
Detect whether a form was sent raw or enhanced

### DISCUSSION THREAD

**Discussion by H14** [sveltejs/kit#6882] [Q&A]
I would like to detect on +page.server.js whether a form has been sent to my server actions with or without progressive enhancement. For example, to find out if certain operations or checks have already been done by the client, or still need to be done by the server. With that I would like to save unnecessary duplicate operations.

Is there a way to find that out from the RequestEvent object that the actions are assigned? 

Or should I craft something myself with (for each form) a hidden field flagged by (non-ssr javascript) as being an enhanced form? Or something like that? 

Thanks!

**Comment by mskocik** [comment]
> should I craft something myself with (for each form) a hidden field flagged by (non-ssr javascript) as being an enhanced formullier

This is the only solution I can imagine and I would go with.

**Comment by nbgoodall** [comment]
Is an `Accept` header check enough for you? E.g. in a server action:

```
if (request.headers.get('accept') === 'application/json') {
  // Enhanced
}
```
